source/app/models/url.rb

```diff
@@ -0,0 +1,31 @@
+class Url < ActiveRecord::Base
+  belongs_to :user
+
+  validates :long_url, presence: true
+  validates :short_url, uniqueness: true
+
+  validate :uri_properly_formatted
+
+  before_create :create_short_url
+
+  def create_short_url
+    begin
+     short_url = SecureRandom.hex(4)
+    end while Url.where(short_url: short_url).exists?
+
+    self.short_url = short_url
+  end
+
+  def compliant?
+    new_uri = URI.parse(self.long_url)
+    new_uri.is_a?(URI::HTTP) && !new_uri.host.nil?
```

Comment:
I thought you had a bug here. Instead, I just learned that `URI::HTTPS` is a subclass of `URI::HTTP`. 👍